misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/PartialCoverageModifiedCounter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.runtime.entity;
+
+import org.jacoco.core.analysis.ICounter;
+
+/**
+ * Represents a counter that eliminates partially covered counter status and coverts it to fully covered status.
+ * @since 2.0.0
+ */
+public class PartialCoverageModifiedCounter implements ICounter {
+
+    private int covered;
+    private int missed;
+
+    public PartialCoverageModifiedCounter(ICounter prevCounter) {
+        this.covered = prevCounter.getCoveredCount();
+        this.missed = prevCounter.getMissedCount();
+        modifyCoverageNumbers();
+    }
+
+    /**
+     * Modify the covered and missed numbers in cases the counter status is calculated as PARTLY_COVERED.
+     * It converts the counter status to FULLY_COVERED.
+     */
+    private void modifyCoverageNumbers() {
+        if (getStatus() == PARTLY_COVERED) {
+            covered = covered + missed;
+            missed = 0;
+        }
+    }
+
+    @Override
+    public double getValue(CounterValue value) {
+        switch (value) {
+            case TOTALCOUNT:
+                return getTotalCount();
+            case MISSEDCOUNT:
+                return getMissedCount();
+            case COVEREDCOUNT:
+                return getCoveredCount();
+            case MISSEDRATIO:
+                return getMissedRatio();
+            case COVEREDRATIO:
+                return getCoveredRatio();
+            default:
+                throw new RuntimeException("No such CounterValue object");
```

Comment:
@azinneera , what is the practice followed in testarina coverage generation to handle errors?

Reply:
We catch the exceptions in RunTestsTask and throw a `BLauncherException` but only the checked exceptions are handled. The original jacoco method itself throws an `AssertionError` which is unchecked. Would it be better to have the `AssertionError` itself?